backend/battling/forms.py

```diff
@@ -8,17 +11,80 @@
     is_team_valid,
     pokemon_in_api,
 )
+from services.email import send_battle_invite
+from users.helper import is_email_valid
 from users.models import User
 
 
+# is_guest = False
+
+
 class CreateBattleForm(forms.ModelForm):
+    opponent = forms.EmailField(
+        required=True,
+    )
+
     class Meta:
         model = Battle
-        fields = ("opponent",)
+        fields = [
+            "creator",
+            "opponent",
+        ]
 
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.fields["opponent"].queryset = User.objects.exclude(id=self.initial["creator_id"])
+        super(CreateBattleForm, self).__init__(*args, **kwargs)
+        self.is_guest = False
+        self.fields["creator"].widget = forms.HiddenInput()
+
+    def clean_opponent(self):
+        opponent_email = self.cleaned_data["opponent"]
+        try:
+            opponent = User.objects.get(email=opponent_email)
+        except User.DoesNotExist:
+            self.is_guest = True
+            opponent = User.objects.create(email=opponent_email)
+            random_password = get_random_string(length=64)
+            opponent.set_password(random_password)
+            opponent.save()
+        return opponent
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        try:
+            opponent = str(cleaned_data["opponent"])
+        except Exception:
+            raise forms.ValidationError("ERROR: Please, type a valid email.")
+
+        if not is_email_valid(opponent):
+            raise forms.ValidationError("ERROR: Please, type a valid email.")
+
+        if cleaned_data["opponent"] == cleaned_data["creator"]:
+            raise forms.ValidationError("ERROR: You can't challenge yourself.")
+
+    def save(self, commit=True):
+        instance = super().save()
+        battle = self.instance
+
+        opponent_team_id = Team.objects.create(battle=battle, trainer=battle.opponent)
+
+        if self.is_guest is False:
+            send_battle_invite(battle, opponent_team_id.id)
+
+        else:
+            invite_form = PasswordResetForm(data={"email": battle.opponent.email})
+            invite_form.is_valid()
+            invite_form.save(
+                self,
+                subject_template_name="registration/password_reset_subject.txt",
+                email_template_name="registration/password_reset_email.html",
```

Comment:
You should use your own templates for subject and email. And they should have different file names and contents.
Make the contents reflect the behavior here, which is an invite, not a password reset.